Add sites table row component to devdocs

wojtekn opened the pull request:
#### Proposed Changes

In this PR, I propose to add the sites table row component to devdocs to let us test different states easier.

#### Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

1. Navigate to `/devdocs/design`
2. Search for SitesTableRow 
3. Click on component name

#### Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to https://github.com/Automattic/wp-calypso/issues/69130

wojtekn committed:
Add SitesTableRow to DevDocs to let us easily test different states

wojtekn committed:
Use absolute path in example

wojtekn commented on client/sites-dashboard/components/sites-table-row/docs/example.jsx:
```diff
@@ -0,0 +1,212 @@
+
+	return (
+		<div>
+			<h2>Free Simple Site</h2>
```
Would it be more readable if we kept all sites in one `<table>`, added `<thead>`, and used the example titles e.g. "Free Simple Site" as the site name?

sejas replied:
I don't have a strong opinion.
I guess it would be simpler to have a unique table.
Then we could have an array of sites, which is more handy to maintain.
```jsx
{ [
	siteSimpleFree,
	siteSimpleBusinessComingSoon,
	siteSimpleBusinessExpired,
	siteJetpackComplete,
	siteSimplePrivateP2,
].map( ( site ) => (
	<SitesTableRow site={ site } key={ site.ID } />
) ) }
```

sejas commented on client/sites-dashboard/components/sites-table-row/docs/example.jsx:
```diff
@@ -0,0 +1,212 @@
+		<div>
+			<h2>Free Simple Site</h2>
+			<div
+				style={ {
+					backgroundColor: 'white',
+					border: '1px solid lightgray',
+					padding: '16px',
+				} }
```
We could abstract this style object to a global variable in this same file.

wojtekn replied:
Thanks, I extracted it to the separate constant.

sejas commented on client/sites-dashboard/components/sites-table-row/docs/example.jsx:
```diff
@@ -0,0 +1,212 @@
+			>
+				<table>
+					<tbody>
+						<SitesTableRow
+							site={ siteSimpleBusinessComingSoon }
+							key={ siteSimpleBusinessComingSoon.ID }
+						></SitesTableRow>
```
Because we are not passing children to the component, we can close the tag earlier.
```suggestion
						<SitesTableRow
							site={ siteSimpleBusinessComingSoon }
							key={ siteSimpleBusinessComingSoon.ID }
						/>
```

wojtekn replied:
Thanks, changed to short form.

sejas reviewed (approved):
Looking good! 👍 
I've added a comment to abstract the style css object that is repeated in each row.

Probably having a unique table will have a better experience and it will be easier to maintain.
However , this is working good and is very explicit, so no strong opinion.

I've noticed that `SiteTableRow` and `SiteThumbnail` are not available on calypso live. And `SiteThumbnail` is not available on wpcalypso neither.
I'm wondering if we need to build/deploy devdocs independently.

wojtekn committed:
Refactor to keep code against DRY rule

wojtekn committed:
Display all rows in the same table to let us test they together

wojtekn commented:
> I don't have a strong opinion.
> I guess it would be simpler to have a unique table.
> Then we could have an array of sites, which is more handy to maintain.

@sejas, thanks for your feedback. I extracted this code and used a `map` as you proposed to avoid repetition and make structural changes easier.

Then I tried changing it to one table, and I think it's better - that way, we can test not only isolated rows but we can test how they work together in the context of table columns.

sejas commented on client/sites-dashboard/components/sites-table-row/docs/example.jsx:
```diff
@@ -0,0 +1,156 @@
+import SitesTableRow from '../../sites-table-row';
+
+export default function SitesTableRowExample() {
+	const CONTAINER_STYLES = {
+		backgroundColor: 'white',
+		border: '1px solid lightgray',
+		padding: '16px',
+	};
+
+	const exampleSite = {
```
@wojtekn , what do you think about moving these static constants (style and sites) outside the component?

sejas reviewed (approved):
Looking great!
Yes, using only one table looks better 👍 

I've added another non-blocking suggestion.

wojtekn committed:
Move constants outside the component

wojtekn committed:
Set displayName for the example component

wojtekn committed:
Fix SiteThumbnail for devdocs/design as well

wojtekn commented:
I've fixed the `SitesTableRow` example definition to include it in the `/devdocs/design` in all environments. While working on it, I fixed the `SiteThumbnail` example as well.

They both were missing the `displayName` property, but this property is required by Devdocs Design.
